### PR TITLE
Maksym/dynamic fees

### PR DESCRIPTION
**Pull request by maksym-springsoftware**
- [Order] add fee currency
- [AccounService] trading fees for instruments - Binance, BinanceFutures, Ftx, Poloniex

### CODE REVIEW

File: xchange-core/src/main/java/org/knowm/xchange/service/trade/TradeService.java [hunk 1]
```diff
@@ -1,9 +1,5 @@
 package org.knowm.xchange.service.trade;
 
-import java.io.IOException;
```
Comment (vasylp3) [inline xchange-core/src/main/java/org/knowm/xchange/service/trade/TradeService.java:3]: Please revert import changes

File: xchange-core/src/main/java/org/knowm/xchange/service/trade/params/TradeHistoryParamOrder.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.knowm.xchange.service.trade.params;
+
+public interface TradeHistoryParamOrder extends TradeHistoryParams {
+	String getId();
```
Comment (vasylp3) [inline xchange-core/src/main/java/org/knowm/xchange/service/trade/params/TradeHistoryParamOrder.java:4]: Minor: rename id to orderId, taking into account that trades are queried

File: xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceAccountService.java [hunk 3]
```diff
@@ -91,22 +92,36 @@ public Object getRawAccountInfo() throws IOException {
...
       throw BinanceErrorAdapter.adapt(e);
     }
   }
 
+  @Override
+  public Map<Instrument, Fee> getDynamicTradingFees(Set<Instrument> instruments) throws IOException {
+    try {
+      BinanceAccountInformation acc = account();
+      Map<Instrument, Fee> tradingFees = new HashMap<>();
+      instruments.forEach(instrument -> tradingFees.put(instrument, getTradingFee(acc)));
```
Comment (vasylp3) [inline xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceAccountService.java:111]: Either 1) exclude instruments that are missing in exchange.getExchangeSymbols() or 2) throw exception when invalid instrument is queried

File: xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxAccountService.java [hunk 4]
```diff
@@ -53,4 +57,23 @@ public AccountInfo getSubaccountInfo(String subaccount) throws IOException {
...
+  @Override
+  public Map<CurrencyPair, Fee> getDynamicTradingFees() throws IOException {
+    Fee fee = getDynamicTradingFee();
+    return exchange.getExchangeSymbols().stream().collect(Collectors.toMap(pair -> pair, pair -> fee));
+  }
+
+  @Override
+  public Map<Instrument, Fee> getDynamicTradingFees(Set<Instrument> instruments) throws IOException {
+    Fee fee = getDynamicTradingFee();
+    return instruments.stream().collect(Collectors.toMap(instrument -> instrument, instrument -> fee));
```
Comment (vasylp3) [inline xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxAccountService.java:70]: Same as in Binance regarding invalid instruments

**Review by vasylp3 — CHANGES_REQUESTED** [review]

File: xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceAccountService.java (hunk 3 above)
Comment (maksym-springsoftware) [reply xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceAccountService.java]: done

File: xchange-core/src/main/java/org/knowm/xchange/service/trade/params/TradeHistoryParamOrder.java (hunk 2 above)
Comment (maksym-springsoftware) [reply xchange-core/src/main/java/org/knowm/xchange/service/trade/params/TradeHistoryParamOrder.java:4]: fixed

**Review by vasylp3 — APPROVED** [review]

**Pull request merged**